### ISSUE TITLE
Error loading mail account

### ISSUE THREAD

**Issue by alex42ka** [owncloud/mail#1161]
### Steps to reproduce
1. Launch the OC-mail app
2. three progress-circles are turning around
3. an error message appears "Fehler beim Laden des ausgewählten Kontos" - just the center progress circle keeps turning
4. email account isn't loaded

### Expected behaviour
As decribed above. Emails should be loaded.

### Actual behaviour
As described above. Email-account isn't loaded.

### Server configuration
**Operating system**: Ubuntu 14.04

**Web server:** Apache

**Database:** MySQL

**PHP version:** PHP 5.5.9-1ubuntu4.14 (cli) (built: Oct 28 2015 01:34:46)

**ownCloud version:** 8.2

**Updated from an older ownCloud or fresh install:** fresh install

**List of activated apps:**

```
Enabled:
  - calendar: 0.8.1
  - contacts: 0.5.0.0
  - direct_menu: 0.4
  - documents: 0.11.0
  - files: 1.2.0
  - files_locking: 
  - files_mv: 0.8.1a
  - files_pdfviewer: 0.7
  - files_sharing: 0.7.0
  - files_texteditor: 2.0
  - files_trashbin: 0.7.0
  - files_versions: 1.1.0
  - files_videoviewer: 0.1.3
  - firstrunwizard: 1.1
  - gallery: 14.2.0
  - mail: 0.2.1
  - notifications: 0.1.0
  - ownnote: 1.05
  - provisioning_api: 0.3.0
  - tasks: 0.8
  - templateeditor: 0.1
  - updater: 0.6
  - user_ldap: 0.7.0
Disabled:
  - activity
  - encryption
  - external
  - files_antivirus
  - files_external
  - user_external
```

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'instanceid' =>'******',
  'passwordsalt' => '********',
  'secret' => '**********',
  'trusted_domains' => 
  array (
    0 => 'example.com',
    1 => 'www.example.com',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'overwrite.cli.url' => 'https://example.com/owncloud',
  'dbtype' => 'mysql',
  'version' => '8.2.0.12',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'dbpassword' => '********',
  'logtimezone' => 'UTC',
  'installed' => true,
  'ldapIgnoreNamingRules' => false,
  'appstore.experimental.enabled' => true,
  'mail_from_address' => 'cloud',
  'mail_smtpmode' => 'php',
  'mail_domain' => 'example2.com',
  'theme' => '',
  'maintenance' => false,
  'preview_libreoffice_path' => '/usr/bin/libreoffice',
  'loglevel' => 2,
  'trashbin_retention_obligation' => 'auto',
  'memcache.local' => '\\OC\\Memcache\\Redis',
  'filelocking.enabled' => 'true',
  'memcache.distributed' => '\\OC\\Memcache\\Redis',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'redis' => 
  array (
    'host' => 'localhost',
    'port' => 6379,
    'timeout' => 0,
    'dbindex' => 0,
  ),
);

```

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP

#### LDAP configuration (delete this part if not used)

```
+-------------------------------+----------------------------------------------------------------------------+
| Configuration                 | s01                                                                        |
+-------------------------------+----------------------------------------------------------------------------+
| hasMemberOfFilterSupport      |                                                                            |
| hasPagedResultSupport         |                                                                            |
| homeFolderNamingRule          |                                                                            |
| lastJpegPhotoLookup           | 0                                                                          |
| ldapAgentName                 | cn=Manager,dc=example2,dc=com                                        |
| ldapAgentPassword             | ***                                                                        |
| ldapAttributesForGroupSearch  | cn;mail                                                                    |
| ldapAttributesForUserSearch   | cn;mail                                                                    |
| ldapBackupHost                |                                                                            |
| ldapBackupPort                |                                                                            |
| ldapBase                      | o=domains,dc=example2,dc=com                                         |
| ldapBaseGroups                | ou=Groups,domainName=example2.com,o=domains,dc=example2,dc=com |
| ldapBaseUsers                 | o=domains,dc=example2,dc=com                                         |
| ldapCacheTTL                  | 600                                                                        |
| ldapConfigurationActive       | 1                                                                          |
| ldapEmailAttribute            | mail                                                                       |
| ldapExperiencedAdmin          | 0                                                                          |
| ldapExpertUUIDGroupAttr       | cn                                                                         |
| ldapExpertUUIDUserAttr        | sn                                                                         |
| ldapExpertUsernameAttr        |                                                                            |
| ldapGroupDisplayName          | cn                                                                         |
| ldapGroupFilter               | (|(objectclass=posixGroup))                                                |
| ldapGroupFilterGroups         |                                                                            |
| ldapGroupFilterMode           | 1                                                                          |
| ldapGroupFilterObjectclass    | posixGroup                                                                 |
| ldapGroupMemberAssocAttr      | memberUid                                                                  |
| ldapHost                      | localhost                                                                  |
| ldapIgnoreNamingRules         |                                                                            |
| ldapLoginFilter               | (&(|(objectclass=mailUser))(|(sn=%uid)))                                   |
| ldapLoginFilterAttributes     | sn                                                                         |
| ldapLoginFilterEmail          | 0                                                                          |
| ldapLoginFilterMode           | 0                                                                          |
| ldapLoginFilterUsername       | 0                                                                          |
| ldapNestedGroups              | 0                                                                          |
| ldapOverrideMainServer        |                                                                            |
| ldapPagingSize                | 500                                                                        |
| ldapPort                      | 389                                                                        |
| ldapQuotaAttribute            |                                                                            |
| ldapQuotaDefault              |                                                                            |
| ldapTLS                       | 0                                                                          |
| ldapUserDisplayName           | cn                                                                         |
| ldapUserFilter                | (|(objectclass=mailUser))                                                  |
| ldapUserFilterGroups          |                                                                            |
| ldapUserFilterMode            | 0                                                                          |
| ldapUserFilterObjectclass     | mailUser                                                                   |
| ldapUuidGroupAttribute        | auto                                                                       |
| ldapUuidUserAttribute         | auto                                                                       |
| turnOffCertCheck              | 0                                                                          |
| useMemberOfToDetectMembership | 0                                                                          |
+-------------------------------+----------------------------------------------------------------------------+

```

### Client configuration
**Browser:** Firefox, Chromium, doesn't matter

**Operating system:** just tested with Ubuntu

#### Web server error log
```
No relevant entries
```

#### ownCloud log (data/owncloud.log)

The log is too long. I will attach it.
[owncloud.log.txt](https://github.com/owncloud/mail/files/37226/owncloud.log.txt)

**Comment by PoPoutdoor** [comment]
@alex42ka Seems your OwnCloud server is not configure properly from the log... please disable ALL apps and fix LDAP first, then try enable Mail app and report back.

**Comment by Gomez** [comment]
I see this error too. Its a 500 Error on this route:

https://xxx/index.php/apps/mail/accounts

No Mail account setup yet.

**Comment by PoPoutdoor** [comment]
I see mail server auth error too.

@alex42ka It is not easy to extracts useful info from a long run log file... please at least post only log entries dated on last test run.

**Comment by Gomez** [comment]
@alex42ka Would you post the php error log? You installed mail from the appstore?

Solved for me, with a clean install.

**Comment by alex42ka** [comment]
Thanks for your answers.

@PoPoutdoor LDAP is configured properly. The log just includes entries of the last month and I've tested some LDAP settings. Here comes just one entry where I tried to access my mail account:

´´´
 Error 	index 	Exception: {"Exception":"Horde_Imap_Client_Exception","Message":"Error connecting to mail server.","Code":11,"Trace":"#0 \/var\/www\/owncloud\/apps\/mail\/vendor\/pear-pear.horde.org\/Horde_Imap_Client\/Horde\/Imap\/Client\/Socket.php(374): Horde_Imap_Client_Socket->_connect()\n#1 \/var\/www\/owncloud\/apps\/mail\/vendor\/pear-pear.horde.org\/Horde_Imap_Client\/Horde\/Imap\/Client\/Base.php(831): Horde_Imap_Client_Socket->_login(*** username and password replaced ***)\n#2 \/var\/www\/owncloud\/apps\/mail\/lib\/account.php(114): Horde_Imap_Client_Base->login(*** username and password replaced ***)\n#3 \/var\/www\/owncloud\/apps\/mail\/lib\/account.php(154): OCA\\Mail\\Account->getImapConnection()\n#4 \/var\/www\/owncloud\/apps\/mail\/lib\/account.php(197): OCA\\Mail\\Account->listMailboxes()\n#5 \/var\/www\/owncloud\/apps\/mail\/lib\/account.php(211): OCA\\Mail\\Account->getMailboxes()\n#6 \/var\/www\/owncloud\/apps\/mail\/lib\/controller\/folderscontroller.php(60): OCA\\Mail\\Account->getListArray()\n#7 [internal function]: OCA\\Mail\\Controller\\FoldersController->index()\n#8 \/var\/www\/owncloud\/lib\/private\/appframework\/http\/dispatcher.php(159): call_user_func_array(Array, Array)\n#9 \/var\/www\/owncloud\/lib\/private\/appframework\/http\/dispatcher.php(89): OC\\AppFramework\\Http\\Dispatcher->executeController(Object(OCA\\Mail\\Controller\\FoldersController), 'index')\n#10 \/var\/www\/owncloud\/lib\/private\/appframework\/app.php(108): OC\\AppFramework\\Http\\Dispatcher->dispatch(Object(OCA\\Mail\\Controller\\FoldersController), 'index')\n#11 \/var\/www\/owncloud\/lib\/private\/appframework\/routing\/routeactionhandler.php(45): OC\\AppFramework\\App::main('FoldersControll...', 'index', Object(OC\\AppFramework\\DependencyInjection\\DIContainer), Array)\n#12 [internal function]: OC\\AppFramework\\routing\\RouteActionHandler->__invoke(Array)\n#13 \/var\/www\/owncloud\/lib\/private\/route\/router.php(282): call_user_func(Object(OC\\AppFramework\\routing\\RouteActionHandler), Array)\n#14 \/var\/www\/owncloud\/lib\/base.php(851): OC\\Route\\Router->match('\/apps\/mail\/acco...')\n#15 \/var\/www\/owncloud\/index.php(39): OC::handleRequest()\n#16 {main}","File":"\/var\/www\/owncloud\/apps\/mail\/vendor\/pear-pear.horde.org\/Horde_Imap_Client\/Horde\/Imap\/Client\/Socket.php","Line":579} 	2015-11-21T16:46:41+00:00 
´´´
I can't deactivate all apps for long, because it is a production system.

**Comment by alex42ka** [comment]
@Gomez Just activated the php error log. No entries after trying to access my mail account.

I've installed the app out of the appstore and it worked flawless for some days. But no this bug prevents me from using it. 

Do you mean that I should download the git version to try it?

**Comment by PoPoutdoor** [comment]
@alex42ka Can you reset the mail account table and try testing with one of the free webmail service account like gmail/outlook.com/yahoo?

**Comment by alex42ka** [comment]
@PoPoutdoor Do you mean that I should just delete the configured account? Or should I delete the database table? If I should do this could you give me some more advice how to do so?

**Comment by alex42ka** [comment]
Tried to delete just the account and added a google account which worked flawless. Then I tried to readd the local email account again and discovered something that maybe has been the problem which prevented the mail app from loading the mail account.
I've discovered that the external ip of the server was banned by fail2ban. I've unbanned this IP and now it is working again as expected. If something happens again I will report it here.
